### ISSUE TITLE
Mixing ASG and Fargate capacity providers does not work

### ISSUE THREAD

**Issue by jcajkovic** [terraform-aws-modules/terraform-aws-ecs#68]
## Description

When trying to use both Fargate and Autoscaling capacity providers I get:

Error: error updating ECS Cluster (dev-main-ecs-cluster) Capacity Providers: InvalidParameterException: A capacity provider strategy cannot contain a mix of capacity providers using Auto Scaling groups and Fargate providers. Specify one or the other and try again.

If your request is for a new feature, please use the `Feature request` template.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: 4.0.2

- Terraform version:
```
1.0.11
```
- Provider version(s):
```
aws = {
    source  = "hashicorp/aws"
    version = "~> 4.0"
}
```

## Reproduction Code [Required]

```
module "ecs" {
  source  = "terraform-aws-modules/ecs/aws"
  version = "~> 4.0.2"

  cluster_name = local.cluster_name

  fargate_capacity_providers = {
    FARGATE = {
      default_capacity_provider_strategy = {
        weight = var.fargate_weight
      }
    }
    FARGATE_SPOT = {
      default_capacity_provider_strategy = {
        weight = var.fargate_spot_weight
      }
    }
  }

  autoscaling_capacity_providers = {
    linux = {
      auto_scaling_group_arn         = module.asg_linux.autoscaling_group_arn
      managed_termination_protection = "DISABLED"

      managed_scaling = {
        maximum_scaling_step_size = 5
        minimum_scaling_step_size = 1
        status                    = "ENABLED"
        target_capacity           = 90
      }

      default_capacity_provider_strategy = {
        weight = 60
        base   = 20
      }
    }
    windows = {
      auto_scaling_group_arn         = module.asg_windows.autoscaling_group_arn
      managed_termination_protection = "DISABLED"

      managed_scaling = {
        maximum_scaling_step_size = 2
        minimum_scaling_step_size = 1
        status                    = "ENABLED"
        target_capacity           = 90
      }

      default_capacity_provider_strategy = {
        weight = 40
      }
    }
  }

}
```

I'm not using workspaces
 
yes I've cleared local chache

### Terminal Output Screenshot(s)

![image](https://user-images.githubusercontent.com/782371/176666977-d8c27fe6-bae2-416a-b783-a9072f9543db.png)

**Comment by rich7021** [comment]
Same here, terraform version 1.2.2, module version 4.1.0.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.1.1 :tada:
